Jenkinsfile

```
@@ -124,23 +130,58 @@ pipeline {
               }
               stashV2(name: 'build', bucket: "${JOB_GCS_BUCKET_STASH}", credentialsId: "${JOB_GCS_CREDENTIALS}")
               archiveArtifacts allowEmptyArchive: true,
-                artifacts: "${BASE_DIR}/elastic-apm-agent/target/elastic-apm-agent-*.jar,${BASE_DIR}/apm-agent-attach/target/apm-agent-attach-*.jar,\
-                      ${BASE_DIR}/apm-agent-attach-cli/target/apm-agent-attach-cli-*.jar,${BASE_DIR}/apm-agent-api/target/apm-agent-api-*.jar,\
-                      ${BASE_DIR}/target/site/aggregate-third-party-report.html",
+                artifacts: "\
+                  ${BASE_DIR}/elastic-apm-agent/target/elastic-apm-agent-*.jar,\
+                  ${BASE_DIR}/apm-agent-attach/target/apm-agent-attach-*.jar,\
+                  ${BASE_DIR}/apm-agent-attach-cli/target/apm-agent-attach-cli-*.jar,\
+                  ${BASE_DIR}/apm-agent-api/target/apm-agent-api-*.jar,\
+                  ${BASE_DIR}/target/site/aggregate-third-party-report.html",
                 onlyIfSuccessful: true
             }
           }
         }
+        /** *
+         * Build on Windows environment
+         */
+        stage('Build Windows') {
+          agent { label 'windows-2019-docker-immutable' }
+          options { skipDefaultCheckout() }
+          environment {
+            JAVA_HOME = "C:\\Users\\jenkins\\.java\\${env.JAVA_VERSION}"
+            PATH = "${env.JAVA_HOME}\\bin;${env.PATH}"
+          }
+          when {
+            beforeAgent true
+            anyOf {
+              expression { return params.windows_ci }
```

This won't trigger the build in PRs or branches. Let's add the `branches` one to be mandatory, although the nightly builds/weekly could drive this too, what are you thoughts?

```suggestion
              expression { return params.windows_ci }
              not { changeRequest() }
```

---

We don´t need to have the Windows parts executed on every merge/run on `main` branch, thus here keeping only opt-in activation is fine.

The plan is to execute it with this parameter enabled through the weekly schedule in #1705 (which might still require some work due to the recent changes).